dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/metadata/MetadataExportService.java

```diff
@@ -99,6 +99,21 @@ void getMetadataAsObjectNodeStream( MetadataExportParams params, OutputStream ou
     Map<Class<? extends IdentifiableObject>, Set<IdentifiableObject>> getMetadataWithDependencies(
         IdentifiableObject object );
 
+    /**
+     * Exports an object including a set of selected dependencies. Only a subset
+     * of the specified export parameters are used for the metadata with
+     * dependencies export. All objects are written to given outputStream
+     *
+     * @param object The {@link IdentifiableObject} to be exported with
+     *        dependencies.
+     * @param params {@link MetadataExportParams}
+     * @param outputStream Streaming target.
+     * @throws IOException
+     */
+    void getMetadataWithDependenciesAsNodeStream( IdentifiableObject object,
```

Comment:
`get` is maybe not the best verb here. I also had trouble with "write" in other exports since it is unclear if we mean write to database (=import) or write to output (=export). So I started naming these `export...`.